File: tests/pages/dashboard_page.py
"""
Dashboard Page Object for HRMS Executive & Employee Home (/).
"""

from playwright.sync_api import Page, expect
from tests.pages.base_page import BasePage
from tests.components.punch_button import PunchButtonComponent


class DashboardPage(BasePage):
    PATH = "/"

    def __init__(self, page: Page):
        super().__init__(page)
        self.punch_button = PunchButtonComponent(page)
        self.role_greeting = page.locator("[data-testid='role-greeting'], h1, h2").first
        self.quick_action_cards = page.locator(".grid a, .grid button").all()

    def get_role_greeting(self) -> str:
        """Returns the greeting message on the dashboard."""
        return self.role_greeting.inner_text().strip() if self.role_greeting.is_visible() else ""

    def click_quick_action(self, action_name: str):
        """Clicks a quick action card by text."""
        card = self.page.locator("a, button").filter(has_text=action_name).first
        expect(card).to_be_visible()
        card.click()
        self.page.wait_for_load_state("domcontentloaded")

File: e2e/pages/DashboardPage.ts
import { Page, Locator, expect } from "@playwright/test";
import { BasePage } from "./BasePage";

export class DashboardPage extends BasePage {
  readonly greeting: Locator;
  readonly nextActions: Locator;
  readonly punchCard: Locator;

  constructor(page: Page, baseURL?: string) {
    super(page, baseURL);
    this.greeting = page.locator('[data-testid="dashboard-greeting"], h2:has-text("Welcome"), h2:has-text("Focus")').first();
    this.nextActions = page.locator('[data-testid="next-actions"]');
    this.punchCard = page.locator('[data-testid="punch-card"], [data-testid="punch-in-btn"]').first();
  }

  async goto(): Promise<void> {
    await this.navigate("/");
  }

  async assertLoaded(): Promise<void> {
    await expect(this.page).toHaveURL(new RegExp(`${this.baseURL}/?$`));
    await expect(this.greeting).toBeVisible({ timeout: 10000 });
  }

  async clickNextAction(actionLabel: string | RegExp): Promise<void> {
    const action = this.page.locator(`[data-testid="next-actions"] a:has-text("${actionLabel}")`).first();
    await expect(action).toBeVisible();
    await action.click();
    await this.page.waitForLoadState("domcontentloaded");
  }
}
